### PR TITLE
Added Rewrite to external URL example

### PR DESCRIPTION
**Pull request by steven-tey**
### Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New Example
- [ ] New feature in existing example

### New Example Checklist

- [ ] 🚀 Link to deployment URL on Vercel
- [ ] 📱 Is it responsive? Are mobile and tablets considered?
- [ ] 📚 `README.md` preview link in PR description (branch)
- [ ] ⚙️ Secrets have instructions for how to set up in the readme
- [ ] 🛫 `npm run new-example` was run to create the example

### CODE REVIEW

File: edge-functions/rewrite-to-external-url/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+https://support-site.vercel.pub/
+
+## How to Use
+
+You can choose from one of the following two methods to use this repository:
+
+### One-Click Deploy
+
+Deploy the example using [Vercel](https://vercel.com?utm_source=github&utm_medium=readme&utm_campaign=next-example):
```
Comment (goncy) [inline edge-functions/rewrite-to-external-url/README.md:26]: Can we update the utm from `next-example` to `vercel-example`? There is a PR from Lee updating those in batch thats why its not updated yet 😓

Comment (steven-tey) [reply edge-functions/rewrite-to-external-url/README.md:26]: Fixing this now!

File: edge-functions/rewrite-to-external-url/pages/_middleware.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  if (hostname === 'support-site.vercel.pub' || hostname === 'localhost:3000') {
+    // rewrites "support-site.vercel.pub" to "https://www.zendesk.com"
+    // note that it's important to check if the URL that you are rewriting to
+    // has trailing slashes or not – or else you'll end up with a redirect loop.
+    // also, note that the URL that you are rewriting to must be a full URL,
+    // and has to include the `www` subdomain if it's present.
+    return NextResponse.rewrite(`https://www.zendesk.com/${pathname}/`)
+  }
+  // perform the rewrite anyways (just for this demo)
+  return NextResponse.rewrite(`https://www.zendesk.com/${pathname}/`)
```
Comment (lfades) [inline edge-functions/rewrite-to-external-url/pages/_middleware.js:16]: I don't think this is needed, looks like an escape hatch if the above fails and it should never fail

File: edge-functions/rewrite-to-external-url/package.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "next": "canary",
+    "react": "18.0.0",
+    "react-dom": "18.0.0"
```
Comment (lfades) [inline edge-functions/rewrite-to-external-url/package.json:12]: Isn't next canary warning about `pages/_middleware.js`? 🤔

**Review by lfades — CHANGES_REQUESTED** [review]
Added some comments, also, lets use TypeScript please 🙏

**Review by lfades — CHANGES_REQUESTED** [review]
@steven-tey I'm talking a look at this example and I don't find it useful enough, it's quite simple and using [Next.js Rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites) is actually a better use case for this. Was this example requested somewhere?

**Comment by okbel** [comment]
Yeah, I think I agree. We should add to the Docs how to rewrite to an external url from middleware -> https://nextjs.org/docs/advanced-features/middleware

**Pull request closed without merging**